src/Microsoft.Azure.WebJobs.Extensions.ServiceBus/Listeners/ServiceBusTriggerExecutor.cs

```diff
@@ -24,9 +24,17 @@ public async Task<FunctionResult> ExecuteAsync(Message value, CancellationToken
             TriggeredFunctionData input = new TriggeredFunctionData
             {
                 ParentId = parentId,
-                TriggerValue = value
+                TriggerValue = value,
+                TriggerDetails = FormatTriggerDetails(value)
             };
             return await _innerExecutor.TryExecuteAsync(input, cancellationToken);
         }
+
+        private string FormatTriggerDetails(Message value)
+        {
+            return $"MessageId: {value.MessageId}, " +
+                $"DeliveryCount: {value.SystemProperties.DeliveryCount}, " +
+                $"EnqueuedTime: {value.SystemProperties.EnqueuedTimeUtc}";
+        }
```

Comment:
One other thing that would be super-useful here would be `LockedUntilUtc`. I just had an issue that we've been debugging for a while and I think the issue is ultimately a too-large PrefetchCount. That means messages sit in the local cache and expire before they are sent to the function. If I could have seen that the lock was expiring soon, that would have helped.